## Question 1

* Install Pipenv
* What's the version of pipenv you installed?
* Use `--version` to find out

`pipenv, version 2024.1.0`

## Question 2

* Use Pipenv to install Scikit-Learn version 1.5.2
* What's the first hash for scikit-learn you get in Pipfile.lock?

`"sha256:03b6158efa3faaf1feea3faa884c840ebd61b6484167c711548fce208ea09445",`

## Question 3

Let's use these models!

* Write a script for loading these models with pickle
* Score this client:

```json
{"job": "management", "duration": 400, "poutcome": "success"}
```

What's the probability that this client will get a subscription? 

* 0.359
* 0.559
* 0.759  `<--`
* 0.959

## Question 4

In [3]:
import requests

url = "http://127.0.0.1:8000/predict"
client = {"job": "student", "duration": 280, "poutcome": "failure"}
requests.post(url, json=client).json()


{'request': {'job': 'student', 'duration': 280, 'poutcome': 'failure'},
 'probability': 0.33480703475511053}

What's the probability that this client will get a subscription?

* 0.335  `<--`
* 0.535
* 0.735
* 0.935

## Question 5

Download the base image `svizor/zoomcamp-model:3.11.5-slim`. You can easily make it by using [docker pull](https://docs.docker.com/engine/reference/commandline/pull/) command.

So what's the size of this base image?

* 45 MB
* 130 MB  `<--`
* 245 MB
* 330 MB

## Question 6

Let's run your docker container!

After running it, score this client once again:

In [5]:
client = {"job": "management", "duration": 400, "poutcome": "success"}
requests.post(url, json=client).json()

{'request': {'job': 'management', 'duration': 400, 'poutcome': 'success'},
 'probability': 0.756743795240796}

What's the probability that this client will get a subscription now?

* 0.287
* 0.530
* 0.757 `<--`
* 0.960